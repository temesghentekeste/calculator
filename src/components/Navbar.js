import { Link } from 'react-router-dom';

const Navbar = () => (
  <nav>
    <h1 data-testid="heading">Math Magicians</h1>
    <ul>
      <li data-testid="home">
        <Link to="/">Home</Link>
      </li>
      <li data-testid="calculator">
        <Link to="/calculator">Calculator</Link>
      </li>
      <li data-testid="quote">
        <Link to="/quote">Quote</Link>
      </li>
    </ul>
  </nav>
);

export default Navbar;
