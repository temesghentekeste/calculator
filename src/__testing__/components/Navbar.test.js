import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Navbar from '../../components/Navbar';
import { BrowserRouter as Router } from 'react-router-dom';

let getByTestId;
let component;

beforeEach(() => {
  component = render(
    <Router>
      <Navbar />
    </Router>
  );

  getByTestId = component.getByTestId;
});

it('renders the Navbar component', () => {
  component = renderer
    .create(
      <Router>
        <Navbar />
      </Router>
    )
    .toJSON();

  expect(component).toMatchSnapshot();
});

it('should contain correct heading', () => {
  const headingEl = getByTestId('heading');
  expect(component.getByText(headingEl.textContent)).toBeInTheDocument();
});

it('should contain correct Home link', () => {
  const homeEl = getByTestId('home');
  expect(component.getByText(homeEl.textContent)).toBeInTheDocument();
});

it('should contain correct Calculator link', () => {
  const calculatorEl = getByTestId('calculator');
  expect(component.getByText(calculatorEl.textContent)).toBeInTheDocument();
});

