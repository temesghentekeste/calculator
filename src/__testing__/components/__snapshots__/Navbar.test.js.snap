// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Navbar component 1`] = `
<nav>
  <h1
    data-testid="heading"
  >
    Math Magicians
  </h1>
  <ul>
    <li
      data-testid="home"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      data-testid="calculator"
    >
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      data-testid="quote"
    >
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
